ddl/delete_range.go

```diff
@@ -279,9 +279,20 @@ func insertJobIntoDeleteRangeTable(ctx sessionctx.Context, job *model.Job) error
 		tableID := job.TableID
 		var indexName interface{}
 		var indexID int64
-		if err := job.DecodeArgs(&indexName, &indexID); err != nil {
+		var partitionIDs []int64
+		if err := job.DecodeArgs(&indexName, &indexID, &partitionIDs); err != nil {
```

Comment:
When modifying table schema, there is a new table info and an old one.
It's easy to get the stale IDs if we get it from the info schema. @shenli

Reply:
@shenli Existing function `getPartitionIDs`.
```go
func getPartitionIDs(table *model.TableInfo) []int64 {
	if table.GetPartitionInfo() == nil {
		return []int64{}
	}
	partitionIDs := make([]int64, 0, len(table.Partition.Definitions))
	for _, def := range table.Partition.Definitions {
		partitionIDs = append(partitionIDs, def.ID)
	}
	return partitionIDs
}
```
Is it clearer.

Reply:
In `drop index`, because the `ddl` of same table is serial executed, so get partition IDs from info schema is ok.
But I remain neutral, get the partition IDs from the info schema or use job arguments are both ok for me. >v<

Reply:
"It's easy to get the stale IDs if we get it from the info schema." Can not understand this. We should guarantee that we will get the correct schema when doing DDL.

Reply:
You can ignore the comment @shenli , I just give a hint on how to achieve this...

> We should guarantee that we will get the correct schema when doing DDL.

Reply:
@ciscoxll Please address the comment. We do not need to put partition IDs in the job argument.

Reply:
@shenli @winkyao 
I think it's OK. The processing here is unified with other places.